### ISSUE TITLE
pdfetch_ONS fails

### ISSUE THREAD

**Issue by daneel6672** [abielr/pdfetch#20]
Simple example to reproduce the error:

`uk.gdp <- pdfetch_ONS(c("ABMI"), "qna")`

Gives error:

`Error in raw$months : $ operator is invalid for atomic vectors`

**Comment by nirguk** [comment]
the issue is due to decomissiong of an old API and introduction of a new one.

The current url generated by pdfetch_ONS returns
```
This API is being decommissioned as part of a suite of work to improve the digital products and services we offer. It is planned to be fully retired on 25/11/2024. If you have any queries please contact us at apiservice@ons.gov.uk.\n
```

**Comment by daneel6672** [comment]
Thanks - do you plan to write a wrapper for the new interface?

**Comment by abielr** [comment]
Apologies for the delay in fixing this. I took a look at their updated API documentation, I'm still finding it challenging to figure out how best to design an update so that the user can easily input a small set of information to get at their time series they want. For example, consider the data at https://www.ons.gov.uk/employmentandlabourmarket/peopleinwork/employmentandemployeetypes/timeseries/lf24/lms. The only series ID on this page is `LF24`, but the URL to download the CSV is https://www.ons.gov.uk/generator?format=csv&uri=/employmentandlabourmarket/peopleinwork/employmentandemployeetypes/timeseries/lf24/lms, which has a bunch of other terms in it. Those terms are admittedly in the URL but leave the user typing a lot of information to get at a single series. The beta API meanwhile seems more complicated and I could not figure out how I would be able to get this time series by calling it. I will likely need to reach to the ONS to get more information.

**Comment by daneel6672** [comment]
Thanks for looking into this. I don't understand why the ONS took this huge step backward with their interface when they should have moved towards producing something more like FRED.

**Comment by wjfisher** [comment]
Yeah, I've stumbled into this also. Why does the ONS have to be so frustrating! Thanks for looking into this
